go/vt/vttablet/tabletserver/tx_pool.go

```diff
@@ -269,7 +277,12 @@ func (tp *TxPool) begin(ctx context.Context, options *querypb.ExecuteOptions, re
 		return "", err
 	}
 
-	conn.txProps = tp.NewTxProps(immediateCaller, effectiveCaller, autocommit)
+	workload := options.GetWorkload()
+	if workload == querypb.ExecuteOptions_UNSPECIFIED {
+		workload = querypb.ExecuteOptions_OLTP
+	}
+	timeout := tp.env.Config().TxTimeoutForWorkload(workload)
```

Comment:
the default is oltp timeout in `tp.env.Config().TxTimeoutForWorkload` , the check can be removed.